lib/segment/src/index/field_index/full_text_index/compressed_posting/compressed_chunks_reader.rs

```diff
@@ -0,0 +1,114 @@
+use bitpacking::BitPacker;
+use common::types::PointOffsetType;
+
+use crate::index::field_index::full_text_index::compressed_posting::compressed_common::{
+    BitPackerImpl, CompressedPostingChunk,
+};
+
+pub struct ChunkReader<'a> {
+    data: &'a [u8],
+    pub chunks: &'a [CompressedPostingChunk],
+    pub reminder_postings: &'a [PointOffsetType],
+    last_doc_id: PointOffsetType,
+}
+
+impl<'a> ChunkReader<'a> {
+    pub fn new(
+        data: &'a [u8],
+        chunks: &'a [CompressedPostingChunk],
+        reminder_postings: &'a [PointOffsetType],
+        last_doc_id: PointOffsetType,
+    ) -> Self {
+        Self {
+            data,
+            chunks,
+            reminder_postings,
+            last_doc_id,
+        }
+    }
+
+    pub fn is_in_postings_range(&self, val: PointOffsetType) -> bool {
+        let in_chunks_range =
+            !self.chunks.is_empty() && val >= self.chunks[0].initial && val <= self.last_doc_id;
+        let in_noncompressed_range = !self.reminder_postings.is_empty()
+            && val >= self.reminder_postings[0]
+            && val <= self.last_doc_id;
+        in_chunks_range || in_noncompressed_range
+    }
+
+    pub fn contains(&self, val: &PointOffsetType) -> bool {
+        if !self.is_in_postings_range(*val) {
+            return false;
+        }
+
+        // find the chunk that may contain the value and check if the value is in the chunk
+        let chunk_index = self.find_chunk(val, None);
+        if let Some(chunk_index) = chunk_index {
+            if self.chunks[chunk_index].initial == *val {
+                return true;
+            }
+
+            let mut decompressed = [0u32; BitPackerImpl::BLOCK_LEN];
+            self.decompress_chunk(&BitPackerImpl::new(), chunk_index, &mut decompressed);
+            decompressed.binary_search(val).is_ok()
+        } else {
+            self.reminder_postings.binary_search(val).is_ok()
+        }
+    }
+
+    pub fn get_chunk_size(
```

Comment:
I propose to change function signature, we don't need to read compressed data here
```suggestion
    pub fn get_chunk_size(
        chunks: &[CompressedPostingChunk],
        data_size: usize,
        chunk_index: usize,
    ) -> usize {
```